Completed File

birizibeg opened the pull request:
contains all info from project proposal

nfarnan reviewed (changes requested):
As a final note, this PR modifies the TEMPLATE_team_description.md file. You needed to copy it and modify the copy. Note that you cannot simply make this change as another commit, because adding all of these changes to the git history could cause conflicts with other student's repos (since this file is shared by all). The easiest solution would be to copy this file to your computer outside of the repository, then delete and refork the repo. If you run into issues, please see me in office hours.

birizibeg commented:
Professor Farnan,

where is the link to your office hours?

thanks,
Birizibe

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Nicholas Farnan ***@***.***>
Sent: Tuesday, September 20, 2022 1:32:38 PM
To: nfarnan/cs1666_organization ***@***.***>
Cc: Gnassingbe, Birizibe ***@***.***>; Author ***@***.***>
Subject: Re: [nfarnan/cs1666_organization] Completed File (PR #3)


@nfarnan requested changes on this pull request.

As a final note, this PR modifies the TEMPLATE_team_description.md file. You needed to copy it and modify the copy. Note that you cannot simply make this change as another commit, because adding all of these changes to the git history could cause conflicts with other student's repos (since this file is shared by all). The easiest solution would be to copy this file to your computer outside of the repository, then delete and refork the repo. If you run into issues, please see me in office hours.

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975629881&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=7GaQ74IZW5UcUwxTydxI5yg7DTxk%2BAQliuLZKsaDT2E%3D&reserved=0>:

>
 ## Canonical game repo URL:

-URLHERE
+[Suburban Rumble Canonical Repo](https://github.com/birizibeg/Suburban-Rumble)


I don't have access to view this repo. You can either set it to public or add my GitHub user to be able to access the repo. It is up to you, publishing your code is not an issue for this class.

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975633220&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=BqJKzLmMLe1qGFD1aELQid%2Fcx9aZn7q7vYuCQRCIsMo%3D&reserved=0>:

>
 ## Stretch Goals

-* GOAL1HERE
-* GOAL2HERE
+* Creating a larger map with more iteration / battles for the user


How much bigger?

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975634281&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=sD19Y06miLOxlgLycI5iGeJ3YjWjJlZpYVez75UZ%2FOU%3D&reserved=0>:

>
 ## Stretch Goals

-* GOAL1HERE
-* GOAL2HERE
+* Creating a larger map with more iteration / battles for the user
+* Items can be used in battle that relate to each character battled


How many items? Example of item functionality?

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975639133&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=1IHu2tMQ5JZ%2BZwivZnTScrSfKgsXr07jVpWHBOPCS7M%3D&reserved=0>:

>
 ## Final Goals

-* XX%: GOAL1HERE
-* XX%: GOAL2HERE
-* XX%: GOAL3HERE
-...
+* 30%: Adjust fighting AI to be more complex, responsive, and easily changeable (this way we can use an AI at Level 1 and reuse it later, just with edited difficulty)
+* 30%: Improve conversational AI to successfully assess the aggressiveness of user-input text; characters respond based on AI evaluated aggressiveness level
+* 20%: Creating a base level that contains, a simple AI for chatting, and an AI for battling
+* 10%: Have the conversation/fighting functional for at least 5 enemies with an arcade-style of progression



OK, we need to revise and rebalance the goals here. I'd suggest:

10% Player can move around a level and interact with (chat and fight) multiple at least 5 AI entities
20% Fighting UI where the player and AI can (define the fighting rules here, how many moves are supported, e.g., punch, kick, etc.; are there special moves? How can the player activate them? etc.)
10% Chat UI where the player can provide full text input to the AI, and receive text reponses. Can also transition to a fight
20% Fight AI that scales in difficulty and the player progresses
20% Chat AI that can analyze the sentiment of user entered text and respond appropriately (with text or move to fight)

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975639746&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WAZLyjwUpER20M9g1wJR50S1PNNQuEDKetwCJQYTZUw%3D&reserved=0>:

>
 ## Midterm Goals

-* GOAL1HERE
-* GOAL2HERE
-...
+* Create one working level with both the conversation and fighting aspects functional
+* AI will pick random choices until the AI teams make an alternative.


What about the fight AI? I'm fine with fight AI being a non-moving dummy, if the player fight ui/moves are implemented.

—
Reply to this email directly, view it on GitHub<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23pullrequestreview-1114209853&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919600085585%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=71p4Gwd0M27771qfPLeV7Rxv58Q5y%2BkfQmSBpImoAxQ%3D&reserved=0>, or unsubscribe<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FANNXV6CQO6546GX3IYQJ6DLV7HYLNANCNFSM6AAAAAAQQISTAY&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919600085585%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=G3WzovDN4Ts07wyOVy%2FdeYw3jg1ZQNjISwgfgDdx%2FDo%3D&reserved=0>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

birizibeg commented:
Can you let me know if I fixed the logistical issues? I deleted my original repo, re-forked, added a new file and made the canonical repo public.

Does that all look good now?
________________________________
From: Gnassingbe, Birizibe ***@***.***>
Sent: Tuesday, September 20, 2022 2:20 PM
To: nfarnan/cs1666_organization ***@***.***>
Subject: Re: [nfarnan/cs1666_organization] Completed File (PR #3)

Professor Farnan,

where is the link to your office hours?

thanks,
Birizibe

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Nicholas Farnan ***@***.***>
Sent: Tuesday, September 20, 2022 1:32:38 PM
To: nfarnan/cs1666_organization ***@***.***>
Cc: Gnassingbe, Birizibe ***@***.***>; Author ***@***.***>
Subject: Re: [nfarnan/cs1666_organization] Completed File (PR #3)


@nfarnan requested changes on this pull request.

As a final note, this PR modifies the TEMPLATE_team_description.md file. You needed to copy it and modify the copy. Note that you cannot simply make this change as another commit, because adding all of these changes to the git history could cause conflicts with other student's repos (since this file is shared by all). The easiest solution would be to copy this file to your computer outside of the repository, then delete and refork the repo. If you run into issues, please see me in office hours.

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975629881&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=7GaQ74IZW5UcUwxTydxI5yg7DTxk%2BAQliuLZKsaDT2E%3D&reserved=0>:

>
 ## Canonical game repo URL:

-URLHERE
+[Suburban Rumble Canonical Repo](https://github.com/birizibeg/Suburban-Rumble)


I don't have access to view this repo. You can either set it to public or add my GitHub user to be able to access the repo. It is up to you, publishing your code is not an issue for this class.

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975633220&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=BqJKzLmMLe1qGFD1aELQid%2Fcx9aZn7q7vYuCQRCIsMo%3D&reserved=0>:

>
 ## Stretch Goals

-* GOAL1HERE
-* GOAL2HERE
+* Creating a larger map with more iteration / battles for the user


How much bigger?

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975634281&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=sD19Y06miLOxlgLycI5iGeJ3YjWjJlZpYVez75UZ%2FOU%3D&reserved=0>:

>
 ## Stretch Goals

-* GOAL1HERE
-* GOAL2HERE
+* Creating a larger map with more iteration / battles for the user
+* Items can be used in battle that relate to each character battled


How many items? Example of item functionality?

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975639133&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=1IHu2tMQ5JZ%2BZwivZnTScrSfKgsXr07jVpWHBOPCS7M%3D&reserved=0>:

>
 ## Final Goals

-* XX%: GOAL1HERE
-* XX%: GOAL2HERE
-* XX%: GOAL3HERE
-...
+* 30%: Adjust fighting AI to be more complex, responsive, and easily changeable (this way we can use an AI at Level 1 and reuse it later, just with edited difficulty)
+* 30%: Improve conversational AI to successfully assess the aggressiveness of user-input text; characters respond based on AI evaluated aggressiveness level
+* 20%: Creating a base level that contains, a simple AI for chatting, and an AI for battling
+* 10%: Have the conversation/fighting functional for at least 5 enemies with an arcade-style of progression



OK, we need to revise and rebalance the goals here. I'd suggest:

10% Player can move around a level and interact with (chat and fight) multiple at least 5 AI entities
20% Fighting UI where the player and AI can (define the fighting rules here, how many moves are supported, e.g., punch, kick, etc.; are there special moves? How can the player activate them? etc.)
10% Chat UI where the player can provide full text input to the AI, and receive text reponses. Can also transition to a fight
20% Fight AI that scales in difficulty and the player progresses
20% Chat AI that can analyze the sentiment of user entered text and respond appropriately (with text or move to fight)

________________________________

In teams/TEMPLATE_team_description.md<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23discussion_r975639746&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919599929350%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WAZLyjwUpER20M9g1wJR50S1PNNQuEDKetwCJQYTZUw%3D&reserved=0>:

>
 ## Midterm Goals

-* GOAL1HERE
-* GOAL2HERE
-...
+* Create one working level with both the conversation and fighting aspects functional
+* AI will pick random choices until the AI teams make an alternative.


What about the fight AI? I'm fine with fight AI being a non-moving dummy, if the player fight ui/moves are implemented.

—
Reply to this email directly, view it on GitHub<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnfarnan%2Fcs1666_organization%2Fpull%2F3%23pullrequestreview-1114209853&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919600085585%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=71p4Gwd0M27771qfPLeV7Rxv58Q5y%2BkfQmSBpImoAxQ%3D&reserved=0>, or unsubscribe<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FANNXV6CQO6546GX3IYQJ6DLV7HYLNANCNFSM6AAAAAAQQISTAY&data=05%7C01%7Cbig14%40pitt.edu%7C2dc775fd30e140b4741508da9b2e1d01%7C9ef9f489e0a04eeb87cc3a526112fd0d%7C1%7C0%7C637992919600085585%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=G3WzovDN4Ts07wyOVy%2FdeYw3jg1ZQNjISwgfgDdx%2FDo%3D&reserved=0>.
You are receiving this because you authored the thread.Message ID: ***@***.***>
